deepchecks/vision/checks/model_evaluation/weak_segments_performance.py

```diff
@@ -0,0 +1,208 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module of weak segments performance check."""
+from collections import defaultdict
+from typing import Any, Callable, Dict, List, Optional
+
+import numpy as np
+import pandas as pd
+import torch
+
+from deepchecks import ConditionCategory, ConditionResult
+from deepchecks.core import CheckResult, DatasetKind
+from deepchecks.core.check_result import DisplayMap
+from deepchecks.core.errors import DeepchecksNotSupportedError, DeepchecksProcessError, DeepchecksValueError
+from deepchecks.tabular import Dataset
+from deepchecks.tabular.context import _DummyModel
+from deepchecks.utils.performance.weak_segment_abstract import WeakSegmentAbstract
+from deepchecks.utils.single_sample_metrics import per_sample_cross_entropy
+from deepchecks.utils.strings import format_number, format_percent
+from deepchecks.vision import Batch, Context, SingleDatasetCheck
+from deepchecks.vision.metrics_utils.iou_utils import per_sample_mean_iou
+from deepchecks.vision.metrics_utils.semantic_segmentation_metrics import per_sample_dice
+from deepchecks.vision.task_type import TaskType
+from deepchecks.vision.utils.image_properties import default_image_properties
+from deepchecks.vision.utils.vision_properties import PropertiesInputType
+
+
+class WeakSegmentsPerformance(SingleDatasetCheck, WeakSegmentAbstract):
+    """Search for segments with low performance scores.
```

Comment:
The docstring should rely that the segmentation is done based on properties
Maybe say something on how the loss is calculated